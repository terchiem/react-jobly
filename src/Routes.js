import React, { useContext } from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import UserContext from './UserContext';

// components
import Home from './Home';
import CompanyList from './CompanyList';
import CompanyPage from './CompanyPage';
import JobList from './JobList';
import Auth from './Auth';
import Profile from './Profile';

/** All routes for the Jobly app */

function Routes() {
  const { token } = useContext(UserContext); // TEMP: switch to context user when implemented

  const loggedInRoutes = (
    <Switch>
      <Route exact path='/'>
        <Home />
      </Route>

      <Route exact path='/companies'>
        <CompanyList />
      </Route>
      
      <Route exact path='/companies/:handle'> 
        <CompanyPage />
      </Route>
      
      <Route exact path='/jobs'> 
        <JobList /> 
      </Route>
      
      <Route exact path='/profile'> 
        <Profile /> 
      </Route>

      { /** Page not found */ }
      <Redirect exact to='/' />
    </Switch>
  );

  const loggedOutRoutes = (
    <Switch>
      <Route exact path='/'>
        <Home />
      </Route>

      <Route exact path='/login'> 
        <Auth /> 
      </Route>
      
      { /** Page not found */ }
      <Redirect exact to='/' />
    </Switch>
  )

  return (token ? loggedInRoutes : loggedOutRoutes);
}

export default Routes;