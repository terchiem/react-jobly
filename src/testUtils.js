import React from 'react';
import UserContext from './UserContext';

const testUser = {
  username: "test",
  first_name: "test",
  last_name: "test",
  email: "test@email.com",
  photo_url: null
}

const testToken = "test";

function UserProvider({ children, currentUser=testUser, token=testToken }) {
  return (
    <UserContext.Provider value={{currentUser, token}}>
      {children}
    </UserContext.Provider>
  )
};

export { UserProvider };