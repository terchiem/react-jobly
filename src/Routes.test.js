import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Routes from './Routes';
import { UserProvider } from './testUtils';

test('renders without crashing', () => {
  render(
    <MemoryRouter>
      <UserProvider>
        <Routes />
      </UserProvider>
    </MemoryRouter>
  );
});

test('matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Routes />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});