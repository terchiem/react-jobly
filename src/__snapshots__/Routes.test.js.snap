// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      Jobly
    </h1>
    <p>
      Welcome back, test!
    </p>
  </div>
</DocumentFragment>
`;
